### ISSUE TITLE
Unable to view file details in the admin backend

### ISSUE THREAD

**Issue by MarinBernard** [gsoules/AvantElements#1]
Hi,

I use AvantElements 2.0.0 with Omeka Classic 2.7.1. When the plugin is enabled, I lose access to file detail pages in the admin UI. Trying to view such a page results in an `Omeka_View_Exception`:

````
Omeka_View_Exception: A current item variable has not been set to this view. in /var/www/html/application/views/helpers/GetCurrentRecord.php:27
Stack trace:
#0 /var/www/html/application/libraries/Zend/View/Abstract.php(349): Omeka_View_Helper_GetCurrentRecord->getCurrentRecord()
#1 /var/www/html/application/libraries/globals.php(1863): Zend_View_Abstract->__call()
#2 /var/www/html/plugins/AvantElements/models/DateValidator.php(8): get_current_record()
#3 /var/www/html/plugins/AvantElements/models/ElementFilters.php(29): DateValidator->hideStartEndYears()
#4 /var/www/html/plugins/AvantElements/AvantElementsPlugin.php(63): ElementFilters->filterDisplayElements()
#5 /var/www/html/application/libraries/Omeka/Plugin/Broker.php(268): AvantElementsPlugin->filterDisplayElements()
#6 /var/www/html/application/libraries/globals.php(324): Omeka_Plugin_Broker->applyFilters()
#7 /var/www/html/application/views/helpers/AllElementTexts.php(158): apply_filters()
#8 /var/www/html/application/views/helpers/AllElementTexts.php(232): Omeka_View_Helper_AllElementTexts->_getElementsBySet()
#9 /var/www/html/application/views/helpers/AllElementTexts.php(298): Omeka_View_Helper_AllElementTexts->_getOutputAsHtml()
#10 /var/www/html/application/views/helpers/AllElementTexts.php(94): Omeka_View_Helper_AllElementTexts->_getOutput()
#11 /var/www/html/application/libraries/Zend/View/Abstract.php(349): Omeka_View_Helper_AllElementTexts->allElementTexts()
#12 /var/www/html/application/libraries/globals.php(2057): Zend_View_Abstract->__call()
#13 /var/www/html/admin/themes/default/files/show.php(18): all_element_texts()
#14 /var/www/html/application/libraries/Omeka/View.php(114): include('/var/www/html/a...')
#15 /var/www/html/application/libraries/Zend/View/Abstract.php(888): Omeka_View->_run()
#16 /var/www/html/application/libraries/Zend/Controller/Action/Helper/ViewRenderer.php(912): Zend_View_Abstract->render()
#17 /var/www/html/application/libraries/Zend/Controller/Action/Helper/ViewRenderer.php(933): Zend_Controller_Action_Helper_ViewRenderer->renderScript()
#18 /var/www/html/application/libraries/Zend/Controller/Action/Helper/ViewRenderer.php(972): Zend_Controller_Action_Helper_ViewRenderer->render()
#19 /var/www/html/application/libraries/Zend/Controller/Action/HelperBroker.php(277): Zend_Controller_Action_Helper_ViewRenderer->postDispatch()
#20 /var/www/html/application/libraries/Zend/Controller/Action.php(527): Zend_Controller_Action_HelperBroker->notifyPostDispatch()
#21 /var/www/html/application/libraries/Zend/Controller/Dispatcher/Standard.php(308): Zend_Controller_Action->dispatch()
#22 /var/www/html/application/libraries/Zend/Controller/Front.php(954): Zend_Controller_Dispatcher_Standard->dispatch()
#23 /var/www/html/application/libraries/Zend/Application/Bootstrap/Bootstrap.php(105): Zend_Controller_Front->dispatch()
#24 /var/www/html/application/libraries/Zend/Application.php(384): Zend_Application_Bootstrap_Bootstrap->run()
#25 /var/www/html/application/libraries/Omeka/Application.php(73): Zend_Application->run()
#26 /var/www/html/admin/index.php(28): Omeka_Application->run()
#27 {main}
````

Disabling the plugin makes those pages accessible again.

**Comment by DBinaghi** [comment]
I cannot reproduce the problem you have, so it might depend on some bug that has been fixed. See my reply to the other issue you've opened.

**Comment by MarinBernard** [comment]
Hi,

You're right: checking out `master` for AvantCommon and AvantElements resolved the issue on Omeka Classic 2.8.

Thanks!

**Comment by DBinaghi** [comment]
@MarinBernard , glad it did work. Should you feel like doing it, you might try the additional features I've submitted to George Soules for his consideration (you'll find the files at my fork of this plugin, for the moment).
